pkg/webui/components/map/index.js

```diff
@@ -34,74 +36,143 @@ Leaflet.Icon.Default.mergeOptions({
   shadowUrl: require('leaflet/dist/images/marker-shadow.png'),
 })
 
-export default class Map extends React.Component {
+const defaultLocation = [52.3, 4.8]
+const tileLayer = (
+  <TileLayer
+    url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+    attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
+  />
+)
+const defaultMinZoom = 7
+
+export default class LocationMap extends React.Component {
   static propTypes = {
-    // Id is a string used to give the map a unique ID.
-    id: PropTypes.string.isRequired,
+    className: PropTypes.string,
+    clickable: PropTypes.bool,
     // LeafletConfig is an object which can contain any number of properties defined by the leaflet plugin and is used to overwrite the default configuration of leaflet.
-    leafletConfig: PropTypes.shape({}),
+    leafletConfig: PropTypes.shape({
+      zoom: PropTypes.number,
+    }),
+    mapCenter: PropTypes.arrayOf(PropTypes.number),
     // Markers is an array of objects containing a specific properties
     markers: PropTypes.arrayOf(
       // Position is a object containing two properties latitude and longitude which are both numbers.
       PropTypes.shape({
-        position: PropTypes.objectOf(PropTypes.number),
+        position: PropTypes.shape({
+          longitude: PropTypes.number,
+          latitude: PropTypes.number,
+        }),
       }),
-    ).isRequired,
+    ),
+    onClick: PropTypes.func,
     // Widget is a boolean used to add a class name to the map container div for styling.
     widget: PropTypes.bool,
   }
 
   static defaultProps = {
     leafletConfig: {},
+    className: undefined,
     widget: false,
+    markers: [],
+    onClick: () => null,
+    mapCenter: undefined,
+    clickable: false,
   }
 
-  getMapCenter(markers) {
-    // This will calculate zoom and map center long/lang based on all markers provided.
-    // Currently it just returns the first marker.
-    // TODO: action (https://github.com/TheThingsNetwork/lorawan-stack/issues/1241)
-    return markers[0]
-  }
+  constructor(props) {
+    super(props)
 
-  createMap(config, id) {
-    this.map = Leaflet.map(id, {
-      ...config,
-    })
+    this.state = {
+      zoomLevel: props.leafletConfig.zoom,
+      mapCenter: defaultLocation,
```

Comment:
What about the location from the marker? Currently the map widget on the overview just stays at the default location, instead of centering on the (first) marker.